app/models/comment.rb

```diff
@@ -1,4 +1,15 @@
 class Comment < ApplicationRecord
-  belongs_to :author
-  belongs_to :post
+  belongs_to :user, foreign_key: 'author_id'
+  belongs_to :post, foreign_key: 'post_id'
+
+  after_create :increment_comment_counter
+  after_destroy :decrement_comment_counter
```

Comment:
[OPTIONAL] Using `after_save` would have been more straightforward and saved you the stress of creating two methods to increment and decrement. But good job using callbacks.

Reply:
Thanks! will look for that in the future!